ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/fragment/AEMArticleItemFragment.java

```diff
@@ -138,21 +155,62 @@ private String injectCss(@NonNull final String content) {
 
     // endregion WebView content
 
+    //region WebClient
+
     private class AEMWebViewClient extends WebViewClient {
         @Nullable
         @Override
         public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
-            if (url.endsWith(".jpg")) {
-                //TODO: Load local Image WebResource
-            } else if (url.endsWith(".css")) {
-                //TODO: Load CSS WebResource
+            String extension = MimeTypeMap.getFileExtensionFromUrl(url);
+
+            if (extension.isEmpty() || mResources == null) {
+                return super.shouldInterceptRequest(view, url);
+            }
+            try {
+
+                String mimeType;
+
+                switch (extension) {
+                    case "jpg":
+                    case "png":
+                    case "bmp":
+                    case "gif":
+                        mimeType = String.format("image/%s", "jpg".equals(extension) ? "jpeg" : extension);
+                        return getResponseFromFile(mimeType, url);
+                    case "css":
+                        mimeType = "text/css";
+                        return getResponseFromFile(mimeType, url);
+                }
+            } catch (FileNotFoundException | NoSuchElementException e) {
+                Timber.d(e);
             }
 
             return super.shouldInterceptRequest(view, url);
         }
+
+        private WebResourceResponse getResponseFromFile(@NonNull String mimeType, @NonNull String url)
+                throws FileNotFoundException, NoSuchElementException {
+
+            Resource resource = Stream.of(mResources)
+                    .filter(r -> r.getUri().toString().equals(url)).findFirst().get();
```

Comment:
this is executing on a background thread, so you can just directly query the database for the specific resource and not deal with needing LiveData or a list of all resources attached to this article.